encode/av1encode.c

```diff
@@ -875,7 +887,43 @@ static int init_va(void)
     va_status = vaInitialize(va_dpy, &major_ver, &minor_ver);
     CHECK_VASTATUS(va_status, "vaInitialize");
     av1_profile = VAProfileAV1Profile0;
-    entryPoint = VAEntrypointEncSliceLP;
+
+    // select entrypoint
+
+    int num_entrypoints = vaMaxNumEntrypoints(va_dpy);
+    VAEntrypoint* entrypoints = malloc(num_entrypoints * sizeof(*entrypoints));
+    if (!entrypoints) {
+        fprintf(stderr, "error: failed to initialize VA entrypoints array\n");
+        exit(1);
+    }
+
+    vaQueryConfigEntrypoints(va_dpy, av1_profile, entrypoints, &num_entrypoints);
+    int support_encode = 0;
+    for (int slice_entrypoint = 0; slice_entrypoint < num_entrypoints; slice_entrypoint++) 
+    {
+        if (requested_entrypoint == -1) {
+            //Select the entry point based on what is avaiable
+            if ((entrypoints[slice_entrypoint] == VAEntrypointEncSlice) ||
+                (entrypoints[slice_entrypoint] == VAEntrypointEncSliceLP)) {
+                support_encode = 1;
+                entryPoint = entrypoints[slice_entrypoint];
+                break;
+            }
+        } else if ((entrypoints[slice_entrypoint] == requested_entrypoint)) {
+            //Select the entry point based on what was requested in cmd line option
+            support_encode = 1;
+            entryPoint = entrypoints[slice_entrypoint];
+            break;
+        }
+    }
+
+    if(entrypoints)
+        free(entrypoints);
+
+    if (support_encode == 0) {
+        printf("Can't find avaiable or request entrypoints for AV1 profiles\n");
```

Comment:
s/avaiable or request/available or requested/

Reply:
Will update it. Thanks for catching this.